src/coord/src/coord.rs

```diff
@@ -5527,15 +5526,12 @@ mod timeline {
             match &self.state {
                 TimestampOracleState::Writing(ts) => {
                     if ts.less_than(&lower_bound) {
-                        self.advance_to = Some(lower_bound.clone());
+                        self.advance_to = Some(ts.step_forward());
                         self.state = TimestampOracleState::Writing(lower_bound);
                     }
                 }
                 TimestampOracleState::Reading(ts) => {
                     if ts.less_than(&lower_bound) {
-                        // This may result in repetition in the case `lower_bound == ts + 1`.
-                        // This is documented as fine, and concerned users can protect themselves.
-                        self.advance_to = Some(lower_bound.clone());
```

Comment:
This seems to remove the property that it makes the prior writes visible at the moment of read, which .. probably results in an unnecessary delay when trying to read writes. Can we discuss why this is the right fix?

Reply:
Sorry, this is just in `fast_forward` so it isn't as problematic as if it were elsewhere. Let me say instead "I do not understand the collected changes, which alter the times at which things are advanced from what I believed to be correct logic to something else".

Reply:
Here we are in the case where the oracle is in the `Reading` state so there can't possibly be any pending writes (whatever writes were at the moment of transition to `Reading` have been handled by appropriately setting `advance_to` in `read_ts`).

Reply:
I see, you're right that we should set `advance_to` here but the original code still was incorrectly setting the state to `Writing(lower_bound)`. I think the invariant that must be true in general is that if `advance_to` is set then it must be set to a timestamp that is strictly less than `Writing(ts)`, because the latter is the timestamp new data will be written at.

I adjusted the PR according to the above. Does it check out for you now?

Reply:
Still nope, sorry! I still don't see the bug that this fixes. Could you explain that more clearly, or potentially the invariant that is not maintained by the rules?

Reply:
Phew! :D